app/sdk/converter/ParserContext.java

```diff
@@ -76,4 +83,69 @@ public void setUserId(String userId) {
     public String getUserId() {
         return userId;
     }
+
+
+    public DateRange getDateRangeForField(int index) {
+        return (DateRange) getSDKDateRangeForIndex(index);
+    }
+
+    public DateRange getDateRangeForField(String fieldName) {
+        return (DateRange) getSDKDateRangeForFieldName(fieldName);
+    }
+
+    public DateTimeRange getDateTimeRangeForField(int index) {
+        return (DateTimeRange) getSDKDateRangeForIndex(index);
+    }
+
+    public DateTimeRange getDateTimeRangeForField(String fieldName) {
+        return (DateTimeRange) getSDKDateRangeForFieldName(fieldName);
+    }
+
+
+    private SDKDateRange getSDKDateRangeForIndex(int index) {
+        SDKDateRange dateRange = getDateTimeRangeIntegerIndexMap().get(index);
+        return dateRange;
+    }
+
+
+    private SDKDateRange getSDKDateRangeForFieldName(String fieldName) {
+        SDKDateRange dateRange = getDateTimeRangeStringIndexMap().get(fieldName);
+        return dateRange;
+    }
+
+
+    public void putDateTimeRange(int index, String fieldName, SDKDateRange sdkDateRange) {
```

Comment:
Why are the setters public? This does not seem like something we want users to be able to manipulate.